packages/xl-ai/src/streamTool/streamTool.ts

```diff
@@ -0,0 +1,96 @@
+import { DeepPartial } from "ai";
+import type { JSONSchema7 } from "json-schema";
+
+export type InvalidOrOk<T> =
+  | {
+      result: "invalid";
+      reason: string;
+    }
+  | { result: "ok"; value: T };
```

Comment:
This is more commonly referred to as an `Result` type:

```ts
export type Result<T, E = Error> =
  | { ok: true; value: T }
  | { ok: false; error: E };
```

Reply:
thanks, refactored (kept `error` as `string` for now instead of a generic `E`)